### PR TITLE
pick issue key from commit title

### PR DESCRIPTION
**Pull request by nsurendran1991**
I fixed JENKINS-61185 issue.
made code changes to make Jenkins to pick issueKeys from commit message and to send build info into Jira Issue.
JiraBuildInfoSenderImpl will use ChangeLogIssueKeyExtractor to get issue keys from commit message.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/service/JiraBuildInfoSenderImpl.java [hunk 1]
```diff
@@ -119,14 +121,20 @@ public JiraSendInfoResponse sendBuildInfo(final JiraBuildInfoRequest request) {
...
+                                        IssueKeyStringExtractor.extractIssueKeys(branch)
+                                                .stream()
+                                                .map(IssueKey::toString)
+                                                .collect(Collectors.toSet()))
+                        .orElseGet(() -> issueKeyExtractor.extractIssueKeys(build));
+        Set<String> CommitIssueKeys = changeLogIssueKeyExtractor.extractIssueKeys(build);
+        if (!CommitIssueKeys.isEmpty()) {
+            branchIssueKeys.addAll(CommitIssueKeys);
+        }
+        return branchIssueKeys;
```
Comment (lenin-atlassian) [inline src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/service/JiraBuildInfoSenderImpl.java:137]: If I understand this correctly, we are now including issue keys from **both the commit messages and the branch name**?

File: src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/service/JiraBuildInfoSenderImpl.java [hunk 2]
```diff
@@ -119,14 +121,20 @@ public JiraSendInfoResponse sendBuildInfo(final JiraBuildInfoRequest request) {
...
+                Optional.ofNullable(request.getBranch())
+                        .map(
+                                branch ->
+                                        IssueKeyStringExtractor.extractIssueKeys(branch)
+                                                .stream()
+                                                .map(IssueKey::toString)
+                                                .collect(Collectors.toSet()))
+                        .orElseGet(() -> issueKeyExtractor.extractIssueKeys(build));
+        Set<String> CommitIssueKeys = changeLogIssueKeyExtractor.extractIssueKeys(build);
+        if (!CommitIssueKeys.isEmpty()) {
```
Comment (lenin-atlassian) [inline src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/service/JiraBuildInfoSenderImpl.java:134]: Can you add unit tests for this new change?

**Comment by lenin-atlassian** [comment]
Can you test this change with a real Jenkins instance?

File: src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/service/JiraBuildInfoSenderImpl.java (hunk 1 above)
Comment (nsurendran1991) [reply src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/service/JiraBuildInfoSenderImpl.java:137]: Yes. Correct.

File: src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/service/JiraBuildInfoSenderImpl.java (hunk 2 above)
Comment (nsurendran1991) [reply src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/service/JiraBuildInfoSenderImpl.java:134]: I have added junit test cases for this change in JiraBuildInfoSenderImplTest. Can you review the same.

**Review by lenin-atlassian — APPROVED** [review]

**Pull request merged**